src/cpp/src/llm/pipeline.cpp

```diff
@@ -157,17 +160,43 @@ static std::unique_ptr<LLMPipelineImplBase> create(
     const ov::genai::Tokenizer& tokenizer,
     const std::string& device,
     const ov::AnyMap& properties,
-    const ov::genai::GenerationConfig& generation_config) {
+    const ov::genai::GenerationConfig& generation_config,
+    const std::filesystem::path& models_path = {}) {
 
     auto properties_without_draft_model = properties;
     auto draft_model_descr = ov::genai::utils::extract_draft_model_from_config(properties_without_draft_model);
+    
     if (draft_model_descr.model != nullptr) {
-        // FIXME: Add support for StatefulSpeculativeLLMPipeline for non-NPU devices for both models.
-        OPENVINO_ASSERT(device == "NPU" || draft_model_descr.device == "NPU",
-            "Stateful Speculative Decoding is expected to be launched when NPU is requested as "
-            "execution device for one or both models.");
-        auto main_model_descr = ov::genai::ModelDesc(model, tokenizer, device, properties_without_draft_model, {}, generation_config);
-        return std::make_unique<StatefulSpeculativeLLMPipeline>(main_model_descr, draft_model_descr);
+        // Extract Eagle3 configuration from draft model properties
+        // Pass models_path for auto-deducing hidden_layers_list from config.json
+        auto eagle_rt_info = ov::genai::speculative_decoding::extract_eagle_mode_from_config(
+            draft_model_descr.properties, 
+            models_path
+        );
+        
+        if (eagle_rt_info.eagle3_mode) {
+            // Eagle3 Speculative Decoding mode
+            OPENVINO_ASSERT(device == "NPU" || draft_model_descr.device == "NPU",
+                "Stateful Eagle3 Speculative Decoding is expected to be launched when NPU is requested as "
+                "execution device for one or both models.");
+            
+            auto main_model_descr = ov::genai::ModelDesc(model, tokenizer, device, 
+                                                        properties_without_draft_model, {}, generation_config);
+            return std::make_unique<StatefulEagle3LLMPipeline>(
+                main_model_descr, 
+                draft_model_descr, 
+                eagle_rt_info.hidden_layers_list
```

Comment:
I think that this information (layers to use for features extraction) should be parsed right in the `StatefulEagle3LLMPipeline` but not earlier..
By the way, isn't this information training-specific? Shouldn't draft model says us which exactly layers to use?